lib/ruby_lsp/ruby_lsp_rails/definition.rb

```diff
@@ -141,6 +180,42 @@ def collect_definitions(name)
           )
         end
       end
+
+      #: ((Prism::SymbolNode | Prism::StringNode) node) -> void
+      def handle_if_unless_conditional(node)
+        call_node = @node_context.call_node
+        return unless call_node
+
+        arguments = call_node.arguments&.arguments
+        return unless arguments
```

Comment:
It looks like most of these methods need to know if there's a surrounding call_node and if there are any arguments.

How about we pull this into `on_call_node_enter` and then pass whatever is necessary through parameters to these methods, so that we can avoid retrieving the arguments in each one of these?

Reply:
It cannot be put in the on_call_node_enter and _then_ pass the parameters to the methods, because this methods needs to be called on symbol node enter to pass the symbol and search for the focused symbol.

But I refactored to deduplicate that code, because all DSL needs argument in their call_node context.